lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,754 @@
+
+  $t->post_ok('/thermostat/heat',
+              json => {temperature => "23°F"});
+
+To do this we're going to override L<Test::Mojo>'s constructor and L<post_ok>
+methods. The constructur will accept a C<base_url>, C<username>, and
+C<password> parameters. The C<post_ok> method will add the C<Authorization>
+header for us.
+
+Because L<Test::Mojo>'s constructor has some shortcuts built into it for
+passing configuration data to Mojolicious applications, we can't invoke
+C<SUPER::new> without removing our custom arguments first:
+
+  package Test::MyApp;
+  use Mojo::Base 'Test::Mojo';
+  use Mojo::Util 'b64_encode';
+
+  has [my @attr = qw/base_url username password/];
+
+  sub new {
+    my $class = shift;
+    my %args  = @_;
+    my %app   = ();
+
+    # Remove custom attributes
+    @app{@attr} = delete @args{@attr};
+
+    my $self = $class->SUPER::new(%args);
+
+    # Set custom attributes
+    $self
+      ->base_url($app{base_url})
+      ->username($app{username})
+      ->password($app{password});
+
+    return $self;
+  }
+
+  sub post_ok {
+    my $self    = shift;
+    my $url     = $self->base_url . shift;
+    my $headers = (ref $_[0] eq 'HASH' ? shift : {});
+    $headers->{Authorization} //= "Basic " .
+      b64_encode join(':' => $self->username, $self->password), '';
+
+    $self->SUPER::post_ok($url, $headers, @_);
+  }
+
+  1;
+
+Our version of C<post_ok> works just like L<Test::Mojo>'s, but it adds the
+C<Authorization> header our web application needs for every call. To use our
+new L<Test::Mojo> subclass:
+
+  use Test::MyApp;
+  use Test::More;
+
+  my $t = Test::MyApp->new(base_url => 'http://myapp.example.com',
+                           username => 'me',
+                           password => 'secretly');
+
+  $t->post_ok('/blender/faster',
+              json => {speed => 'frappé'})
+    ->status_is(200);
+
+  # Use invalid password
+  $t->password('sorry!');
+  $t->post_ok('/blender/faster',
+              json => {speed => 'frappé'})
+    ->status_is(401);
+
+  done_testing();
+
+Great! Our little L<Test::Mojo>-based class allows us to make authorized
+requests to our deployed web application, and the test assertion (C<post_ok>)
+preserves compatibility with L<Test::Mojo>.
+
+=head1 MORE
+
+You can continue with L<Mojolicious::Guides> now or take a look at the
+L<Mojolicious wiki|http://github.com/kraih/mojo/wiki>, which contains a lot
+more documentation and examples by many different authors.
+
+See also L<Test::Mojo>, L<Mojo::UserAgent>, L<Mojo::DOM>, L<Mojo::JSON>,
+L<Mojo::JSON::Pointer>, and L<Mojo::Transaction> for additional API
+information referenced in this tutorial.
+
+=head1 SUPPORT
+
+If you have any questions the documentation might not yet answer, don't
+hesitate to ask on the
+L<mailing list|http://groups.google.com/group/mojolicious> or the official IRC
+channel C<#mojo> on C<irc.perl.org>
+(L<chat now!|https://chat.mibbit.com/?channel=%23mojo&server=irc.perl.org>).
+
+=cut
```

Comment:
WOW!!!  Seriously, this guide is SOOO fantastic!  You did a great job, thank you so much!  I hope it gets approved and brought into core!!!  :D